### PR TITLE
sys/stdio: add optional function stdio_available

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR provides the definition of an additional `stdio` function to check how many bytes are available for reading.

A number of `stdio` backend implementations allow to check the number of bytes available for reading before calling the blocking `stdio_read`. This helps to implement non-blocking functionalities while waiting for `stdin`.

This PR provides the implementation for
```
stdio_uart
stdio_cdc_acm
```

### Testing procedure

Green CI.

### Issues/PRs references

Follow-up of PR #17335
Required for PR #17447

### CODE REVIEW

File: sys/include/stdio_base.h [hunk 1]
```diff
@@ -34,6 +36,18 @@ extern "C" {
  */
 void stdio_init(void);
 
+#if IS_USED(MODULE_STDIO_AVAILABLE) || DOXYGEN
+/**
+ * @brief   Get the number of bytes available for reading from stdio.
+ *
+ * @warning This is function is only available if the implementation supports
```
Comment (aabadie) [inline sys/include/stdio_base.h:43]:
```suggestion
 * @warning This function is only available if the implementation supports
```

Comment (gschorcht) [reply sys/include/stdio_base.h:43]: Should I squash it directly to avoid an additional cycle in Murdock?

Comment (aabadie) [reply sys/include/stdio_base.h:43]: Yes, please squash!

**Review by aabadie — APPROVED** [review]
ACK

**Comment by gschorcht** [comment]
@aabadie Thanks for reviewing and merging.

**Pull request merged**